docs/signaling.md

```diff
@@ -16,3 +16,43 @@ As we're in a P2P network, it is possible to add bad data. However, we wanted to
 
 * [SignalingMessage.recipientSessionId] and [SignalingMessage.senderSessionId] IDs are overwritten before they go to the TomP2P layer, since they could be faked by WebSocket clients
 * [ClientService.removeClient] only accepts an [IInternalClient] (that you got from [ClientService.addClient]), so you cannot easily disconnect another user.
+
+### Messaging Format
+
+The [Message] class uses a `type` as it's discriminator. Every other field depends on the type. This way, it's a very flexible format that's also easy to cast in a type-safe way.
+
+### Bootstrapping
+
+The [ClientService] class has two bootstrapping constructors: one for bootstrapping with a given TomP2P `PeerAddress`, one with a normal IP/port pair.
+
+* The `PeerAddress` bootstrap mechanism is meant for tests, when you already have the peer's address in the correct, technical format
+* The IP/port pair bootstrap mechanism is meant for user input
+
+### ClientService
+
+The [ClientService] class is meant as the one-stop starting point for all DHT operations. It's meant as a singleton that creates and bootstraps the TomP2P peer. All objects are created through methods of this class.
+
+### Futures
+
+When we decided to build a nicer API for TomP2P Futures, we split it up.
+
+In this layer, you can find the high-level [Future]s. These are a rework of the API that TomP2P gives.
+
+In the [util][ch.hsr.dsl.dwrtc.util] layer, you can find the extension classes that build right on top of TomP2P. Our [Future] classes build on these.
```

Comment:
I'm also not sure if we want to write something like "you can find". Shouldn't it be from a third person perspective? e.g. [util][ch.hsr.dsl.dwrtc.util] contains the extension classes that build right on top of TomP2P an is used by [Future].